### PR TITLE
dladm_vnic python3 fixes and tests

### LINKED ISSUE #131: dladm_vnic: error setting vlan when using python3
##### SUMMARY

Errors trying to set VLAN in `dladm_vnic` module.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

`dladm_vnic`

##### ANSIBLE VERSION

```paste below
ansible 2.10.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/jbronn/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/jbronn/.local/lib/python3.8/site-packages/ansible
  executable location = /home/jbronn/.local/bin/ansible-playbook
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

OmniOS CE r151034t (Illumos); target host Python is 3.7.5.

##### STEPS TO REPRODUCE

Running the following playbook targeting an Illumos host (OmniOS CE):

```yaml
---
- name: 'create a vnic on vlan 23'
  hosts: all
  become: true
  tasks:
    - dladm_vnic:
        name: vnic0
        link: igb0
        vlan: 23
  vars:
    ansible_python_interpreter: '/usr/bin/python3'
```

##### EXPECTED RESULTS

Expect `vnic0` to be created over `igb0` with on VLAN 23 ("VID" in Solaris parlance).

##### ACTUAL RESULTS

```paste below
Traceback (most recent call last):
  File "/export/home/jbronn/.ansible/tmp/ansible-tmp-1602200001.221015-1447844-199957029738156/AnsiballZ_dladm_vnic.py", line 102, in <module>
    _ansiballz_main()
  File "/export/home/jbronn/.ansible/tmp/ansible-tmp-1602200001.221015-1447844-199957029738156/AnsiballZ_dladm_vnic.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/export/home/jbronn/.ansible/tmp/ansible-tmp-1602200001.221015-1447844-199957029738156/AnsiballZ_dladm_vnic.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.community.network.plugins.modules.dladm_vnic', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib/python3.7/runpy.py", line 205, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_dladm_vnic_payload_pf9t6364/ansible_dladm_vnic_payload.zip/ansible_collections/community/network/plugins/modules/dladm_vnic.py", line 260, in <module>
  File "/tmp/ansible_dladm_vnic_payload_pf9t6364/ansible_dladm_vnic_payload.zip/ansible_collections/community/network/plugins/modules/dladm_vnic.py", line 221, in main
  File "/tmp/ansible_dladm_vnic_payload_pf9t6364/ansible_dladm_vnic_payload.zip/ansible_collections/community/network/plugins/modules/dladm_vnic.py", line 183, in is_valid_vlan_id
TypeError: '<=' not supported between instances of 'int' and 'str'
```

### PR DESCRIPTION
**Pull request by jbronn**
##### SUMMARY

Fixes #131 and adds tests for `dladm_vnic` module.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

`dladm_vnic`

##### ADDITIONAL INFORMATION

N/A.

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Cannot judge the code changes, but yay for tests! :)

File: plugins/modules/network/illumos/dladm_vnic.py [hunk 1]
```diff
@@ -176,22 +181,22 @@ def is_valid_unicast_mac(self):
 
         mac_re = re.match(self.UNICAST_MAC_REGEX, self.mac)
 
-        return mac_re is None
+        return mac_re is not None
 
     def is_valid_vlan_id(self):
 
-        return 0 <= self.vlan <= 4095
+        return 0 < self.vlan <= 4095
```
Comment (felixfontein) [inline plugins/modules/network/illumos/dladm_vnic.py:188]: This is the only change I cannot really judge and would wait for a maintainer to look at - except there seem to be no maintainers. Do you have a link to some docs which say that 0 is not a valid VLAN ID?

Comment (jbronn) [reply plugins/modules/network/illumos/dladm_vnic.py]: I also had the upper range wrong and fixed to exclude 4095 as well in f1a95c8.

Here are some reasons that it's *now* correct:
* Trying to create a vnic with a VID of 0/4095 on my physical Illumos host (OmniOS) emits an error about it being unacceptable:
    ```
    sudo dladm create-vnic -t -l net0 -v 0 test0
    dladm: invalid VLAN identifier '0'

    sudo dladm create-vnic -t -l net0 -v 4095 test0
    dladm: invalid VLAN identifier '4095'
    ```
* The Illumos source code for [`dladm-vnic`](https://github.com/illumos/illumos-gate/blob/master/usr/src/cmd/dladm/dladm.c#L4716-L4717) confirms this logic and behavior.
* A VID of 0 is a reserved part of [802.1Q](https://en.wikipedia.org/wiki/IEEE_802.1Q); you'll typically only see 0-tagged frames in the context of [802.1p](https://en.wikipedia.org/wiki/IEEE_P802.1p).

File: changelogs/fragments/131_dladm_vnic.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - dladm_vnic - fixed issue where setting vlan in Python 3 caused a type error (https://github.com/ansible-collections/community.network/issues/131).
```
Comment (felixfontein) [inline changelogs/fragments/131_dladm_vnic.yml:3]:
```suggestion
  - dladm_vnic - fixed issue where setting vlan in Python 3 caused a type error (https://github.com/ansible-collections/community.network/issues/131).
  - dladm_vnic - vlan IDs 0 and 4095 are now correctly identified as invalid (https://github.com/ansible-collections/community.network/pull/132).
```

**Review by felixfontein — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@jbronn thanks for fixing this!

**Comment by Andersson007** [comment]
@jbronn hi, would you like to have your GH login added to `.github/BOTMETA.yml` as a maintainer of the module?
You'll be notified about Issues/PRs related to this file and your `shipit` will be counted by bot for automerge (needs two for bugfixes and minor changes).
What do you think?

**Pull request merged**